### ISSUE TITLE
Creating and training your own model

### ISSUE THREAD

**Issue by nival999** [edufoursans/reading-in-the-dark#1]
Hi,

I have been trying to contribute some code to this project that would enable humans to train new classifiers  with this technique. I have created a series of Jupyter notebooks here that you can read: https://github.com/M-Usman10/FaceRecognition/blob/master/Training.ipynb

 For example, I use this code to create a quadratic polynomial classifier: 

```
weights=tf.Variable(w,dtype=tf.float32)
x=tf.placeholder(dtype=tf.float32,shape=(None,X.shape[1]))
target=tf.placeholder(dtype=tf.float32,shape=(None,1))
b=tf.Variable(0,dtype=tf.float32)
q_polynomial=tf.matmul(x,weights)+b
loss=tf.reduce_mean(tf.nn.sigmoid_cross_entropy_with_logits(logits=q_polynomial, labels=target))
learning_rate = 0.1
batch_size = 1000
iter_num = 1500
optimizer=tf.train.AdamOptimizer(learning_rate)
train_op=optimizer.minimize(loss)
prediction = tf.round(tf.sigmoid(q_polynomial))
init = tf.global_variables_initializer()
```

These are based on instructions from the paper. The only thing it doesn't account for is the integer scaling and rounding (yet). However, when I use tensorflow to save the weights file, it is 2583x1 array, whereas the model loaded inside this repository seems to be a dictionary of a bunch of arrays or matrices. I am trying to decipher what they mean. "40" means 4 columns for each digit? Like 4 columns because it's quadratic? Or.... ?  

I guess I was wondering if someone who is responsible for this repository could enlighten me on how I can turn a quadratic polynomial model from tensorflow into the model structure loaded from the file in this repository. Even if it were only pseudocode or a fork of my repository or a few lines in this repository's readme, that would help tremendously.

**Comment by edufourfr** [comment]
Hi everyone,

Thanks for alerting me to the issue Kevin, I would have missed it. I'm watching the repo now.

Nival, thanks for your interest. Unless I'm missing something the model your are currently describing seems to be linear (your logits are defined by wx+b). If so, there are cryptographic schemes for linear models that are directly suited to what you're trying to do and will be much more efficient. You want to look for implementations of ABDP (https://eprint.iacr.org/2015/017.pdf) or variants thereof.

If I misunderstood your model or you have more questions please feel free to follow up.
